Bio/PDB/Chain.py

```diff
@@ -119,6 +120,10 @@ def __contains__(self, id):
          - id - (string, int, string) or int
 
         """
+        try: 
```

Comment:
I'd include this attempted conversion in the `_translate_id` method and remove the `isinstance` check. This should solve your problem no?